### PR TITLE
Added mime type checker

### LINKED ISSUE #19: Use correct way to load files
Current way of loading files is bad.
Investigate, how to load files correct.

Download attached file, unzip and upload it as thumbnail. Try to open loaded image in certain tab

[cat.php.zip](https://github.com/Madlexx/symfonyApplication/files/166974/cat.php.zip)

### PR DESCRIPTION
**Pull request by Madlexx**
fixed #19

### CODE REVIEW

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 1]
```diff
@@ -23,6 +23,10 @@ public function __construct($uploadDirectory, $absPath)
      */
     public function save(UploadedFile $uploadedFile)
     {
+        if (!in_array($uploadedFile->getMimeType(), [IMAGETYPE_JPEG, IMAGETYPE_PNG])) {
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: I'm trying to load image. `getMimeType` returns `image/jpeg`.
`IMAGETYPE_JPEG ` and `IMAGETYPE_PNG` is `int` value.

Also, my "suspicious" file from issue also has `image/jpeg` mime type.

This solution isn't working.

Comment (Madlexx) [inline src/Application/BlogBundle/Service/Thumbnail.php]: Changed on the extension file checking.

**Comment by aivus** [comment]
Please, see http://symfony.com/doc/current/cookbook/controller/upload_file.html

**Comment by Madlexx** [comment]
Added hash.

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 2]
```diff
@@ -23,15 +23,20 @@ public function __construct($uploadDirectory, $absPath)
      */
     public function save(UploadedFile $uploadedFile)
     {
+        if (!in_array($uploadedFile->getClientOriginalExtension(), ['png', 'jpeg', 'jpg'])) {
+            return false;
+        }
+
         $this->createFolderIfNotExists();
 
+        $name = hash_file('md5', $uploadedFile) . '.' . $uploadedFile->getClientOriginalExtension();
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: `hash_file` is to heavy

**Comment by Madlexx** [comment]
Changed.

**Comment by aivus** [comment]
I made some entity cleanup and little refactor.
File upload is ok for now, but validation should be moved to validator in the future. I have added todo for this.

**Pull request merged**